include/irr/asset/IAssetLoader.h

```diff
@@ -155,6 +180,16 @@ class IAssetLoader : public virtual core::IReferenceCounted
         }
 
         //! Called before loading a file
+		/**
+			\param inOutFilename is a path to file Asset will be loaded from.
```

Comment:
the filename argument goes in, the actual resolved path comes out (the override decides how to resolve a local path or even a URL into a "proper" filename).

Hierarchy level is more like a stack counter, we expect that that asset loading will get called recursively, example mesh needs material, material needs texture, etc. so `getLoadFilename` could be called separately for each dependent resource from deeper recursions in the loading stack.

Reply:
status?

Reply:
Solved